drivers/SmartThings/zigbee-button/src/aqara/init.lua

```diff
@@ -0,0 +1,128 @@
+local capabilities = require "st.capabilities"
+local data_types = require "st.zigbee.data_types"
+local cluster_base = require "st.zigbee.cluster_base"
+local battery_defaults = require "st.zigbee.defaults.battery_defaults"
+local clusters = require "st.zigbee.zcl.clusters"
+local PowerConfiguration = clusters.PowerConfiguration
+
+local PRI_CLU = 0xFCC0
+local PRI_ATTR = 0x0009
+local MFG_CODE = 0x115F
+
+local ROTATE_CLU = 0x000C
+local EVENT_CLU = 0x0012
+local FACE_ATTR = 0x0149
+local ACTION_ATTR = 0x0055
+local CUBE_MODE = 0x0148
+
+local cubeAction = capabilities["stse.cubeAction"]
+local cubeFace = capabilities["stse.cubeFace"]
+local cubeFaceVal = { "face1Up", "face2Up", "face3Up", "face4Up", "face5Up", "face6Up" }
+local cubeFlipToSideVal = { "flipToSide1", "flipToSide2", "flipToSide3", "flipToSide4", "flipToSide5", "flipToSide6" }
+
+local CUBEACTION_TIMER = "cubeAction_timer"
+local CUBEACTION_TIME = 3
+
+local FINGERPRINTS = { mfr = "LUMI", model = "lumi.remote.cagl02" }
+
+local is_aqara_products = function(opts, driver, device, ...)
+  return device:get_manufacturer() == FINGERPRINTS.mfr and device:get_model() == FINGERPRINTS.model
+end
+
+local callback_timer = function(driver, device, cmd)
+  return function()
+    device:emit_event(cubeAction.cubeAction("noAction"))
+  end
+end
+
+local function delete_timer(device)
+  local timer = device:get_field(CUBEACTION_TIMER)
+  if timer then
+    device.thread:cancel_timer(timer)
+    device:set_field(CUBEACTION_TIMER, nil, { persist = true })
+  end
+end
+
+local function data_handler(driver, device, value, zb_rx)
+  local val = value.value
+  local timer = nil
+  if val == 0x0000 then -- Shake
+    device:emit_event(cubeAction.cubeAction("shake"))
+    delete_timer(device)
+    device:set_field(CUBEACTION_TIMER, device.thread:call_with_delay(CUBEACTION_TIME, callback_timer(driver, device)))
```

Comment:
These two lines always appear together. I would suggest you pull all of this logic out into one function rather than defining two functions that are always called together in the same order.

Reply:
I fixed it.